src/rabbit_mqtt_processor.erl

```diff
@@ -463,17 +463,8 @@ process_login(UserBin, PassBin, ProtoVersion,
                                   adapter_info = set_proto_version(AdapterInfo, ProtoVersion)}) of
         {ok, Connection} ->
             case rabbit_access_control:check_user_loopback(UsernameBin, Sock) of
-                ok          ->
-                  {ok, User} = rabbit_access_control:check_user_login(
-                                 UsernameBin,
-                                 case PassBin of
-                                   none -> [];
-                                   P -> [{password,P}]
-                                 end),
-                  {?CONNACK_ACCEPT, Connection, VHost, #auth_state{
-                                                         user = User,
-                                                         username = UsernameBin,
-                                                         vhost = VHost}};
+                ok          -> [{internal_user, InternalUser}] = amqp_connection:info(Connection, [internal_user]),
```

Comment:
No exception handling added here because if it cannot retrieve the internal user info, it means another bigger problem.